pkg/v1/remote/transport/bearer.go

```diff
@@ -67,24 +69,48 @@ func stringSet(ss []string) map[string]struct{} {
 
 // RoundTrip implements http.RoundTripper
 func (bt *bearerTransport) RoundTrip(in *http.Request) (*http.Response, error) {
-	sendRequest := func() (*http.Response, error) {
+	sendRequest := func(registryToken string) (*http.Response, error) {
 		// http.Client handles redirects at a layer above the http.RoundTripper
 		// abstraction, so to avoid forwarding Authorization headers to places
 		// we are redirected, only set it when the authorization header matches
 		// the registry with which we are interacting.
 		// In case of redirect http.Client can use an empty Host, check URL too.
 		if matchesHost(bt.registry, in, bt.scheme) {
-			hdr := fmt.Sprintf("Bearer %s", bt.bearer.RegistryToken)
+			hdr := fmt.Sprintf("Bearer %s", registryToken)
 			in.Header.Set("Authorization", hdr)
 		}
 		return bt.inner.RoundTrip(in)
 	}
 
-	res, err := sendRequest()
-	if err != nil {
-		return nil, err
+	var res *http.Response
+	var err error
```

Comment:
I think we can define this inside the loop, and change `return res, err` -> `return res, nil` on line 109. We'll either get an `err` and `return nil, err`, or we'll get a good `res` and `return res, nil`, and restructuring in this way makes this clearer at a glance.

Reply:
fixed